WordPress/gutenberg
# rich-text: set up autogenerated API docs

**Related to** https://github.com/WordPress/gutenberg/issues/14227

This PR sets up auto-generated API docs for the `rich-text` package through the `npm run docs:build` command and updates its README file.

## Notable modifications

- Fixed optional parameters. Should be `{type} [param]`, not `{?type} param`.
- Fixed returns when type can be undefined. Should be `{type|undefined}`, not `{?type}`.
- New symbols added that were undocumented:
  - `insertObject`
  - `unregisterFormatType`
- Symbols that were documented and are removed in this PR: `apply`.
- Symbols that stay undocumented:
  - `changeListType`
  - `charAt`
  - `getSelectionEnd`
  - `getSelectionStart`
  - `indentListItems`
  - `insertLineBreak`
  - `insertLineSeparator`
  - `isEmptyLine`
  - `LINE_SEPARATOR`
  - `outdentListItems`
  - `unstableToDom`

## Test instructions

- Delete or modify the contents at `packages/rich-text/README.md` within the start `` and end `` tokens.
- Execute `npm run docs:build` and verify they are re-created.
- Review that the API docs match the JSDoc of the exported symbols.
